test/Microsoft.AspNet.Razor.Test/TagHelpers/TagHelperParseTreeRewriterTest.cs

```diff
@@ -19,6 +19,367 @@ namespace Microsoft.AspNet.Razor.Test.TagHelpers
     public class TagHelperParseTreeRewriterTest : TagHelperRewritingTestBase
     {
         [Fact]
+        public void Rewrite_RecoversWhenRequiredAttributeMismatchAndRestrictedChildren()
```

Comment:
Yup down in the Theory test the p and strong tags both have several levels of `allowedChildren`